File: Exercise.java
import java.util.Scanner;
import java.util.Random;

public class Exercise {
    // O(n^3)
    public static void cubicOperation(int[][][] threeDimensionalArray) { // O(1)
        int n = threeDimensionalArray.length; // O(1)

        for (int i = 0; i < n; i++) { // O(n)
            for (int j = 0; j < n; j++) { // O(n)
                for (int k = 0; k < n; k++) { // O(n)
                    threeDimensionalArray[i][j][k]++; // O(1)
                }
            }
        }
    }

    public static void main(String[] args) {
        System.out.println("Enter dimension size (n): ");
        int n;
        try (Scanner scanner = new Scanner(System.in)) {
            n = scanner.nextInt();
        }

        int[][][] threeDimensionalArray = new int[n][n][n];
        Random random = new Random();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    threeDimensionalArray[i][j][k] = random.nextInt(n);
                }
            }
        }

        long startTime = System.nanoTime();

        cubicOperation(threeDimensionalArray);

        long endTime = System.nanoTime();

        double runtimeNS = endTime - startTime;
        double runtimeS = runtimeNS / 1000000000.0;

        System.out.println("Runtime in ns: " + runtimeNS);
        System.out.println("Runtime in s: " + runtimeS);
    }
}

File: Exercise.py
from typing import List
import random
import time

def perform_cubic_operation(three_dimensional_array: List[List[List[int]]]): # O(1)
    n = len(three_dimensional_array) # O(1)

    for i in range(n): # O(n)
        for j in range(n): # O(n)
            for k in range(n): # O(n)
                three_dimensional_array[i][j][k] += 1 # O(1)

def read_input():
    while True:
        try:
            return int(input("Enter dimension size (n): "))
        except ValueError:
            print("Please enter a number.")
        except KeyboardInterrupt:
            exit()

def main():
    n = read_input()

    three_dimensional_array = [[[random.randint(1, 100) for _ in range(n)] for _ in range(n)] for _ in range(n)]

    start_time = time.perf_counter()

    perform_cubic_operation(three_dimensional_array)

    end_time = time.perf_counter()
    duration = end_time - start_time

    print(f"Execution time: {duration:.4f} seconds.")

if __name__ == "__main__":
    main()